### ISSUE TITLE
Error using pi in a Data Definition

### ISSUE THREAD

**Issue by bmaclach** [JacquesCarette/Drasil#1099]
I am trying to add a DD to the SSP example which contains pi as part of the equation. Here is the equation from the manual version:
![image](https://user-images.githubusercontent.com/18634568/51924206-9a3ba480-23ba-11e9-8864-1674c7c5469c.png)

The code I am using for the `Expr` is:
```
ratioVarEqn = case_ [case1, case2]
  where case1 = (1, (sy constF))

        case2 = (sin ((sy QM.pi_) * (((inxi slipDist) - (idx (sy slipDist) 0)) /
                ((indxn slipDist) - (idx (sy slipDist) 0)))), UnaryOp Not (sy constF))
```

where `pi_` is a `QuantityDict` imported from Data.Drasil.Quantities.Math. When I try to compile this, I get the following error:
```
mkdir -p build/SSP
cd build/SSP && stack exec -- ssp
ssp.EXE: Concept: pi not found in ConceptMap
CallStack (from HasCallStack):
  error, called at .\Language\Drasil\ChunkDB.hs:177:24 in drasil-lang-0.1.54-4diRkNt3Fw3EmwDBs7fCV4:Language.Drasil.ChunkDB
make: *** [ssp_build] Error 1
```
I'm not certain, but my guess is that a Data Definition expects all of its symbols to be concepts, so that it can display the description for each symbol in the description field of the Data Definition, and since pi does not have a description, it throws an error.

I think this is a bug but I'm not certain, it could be that I'm just missing some extra step I need to take in order to use pi in a DD.

**Comment by Mornix** [comment]
It looks like `pi_` is missing from the `ChunkDB`s in SSP's [Body.hs](https://github.com/JacquesCarette/Drasil/blob/6ea7593312e52416e651ede5fd38836c3f52c212/code/drasil-example/Drasil/SSP/Body.hs#L195). If you add `pi_` to the DB, does that fix the error? 

(You may need to also add it to [`usedDB`](https://github.com/JacquesCarette/Drasil/blob/master/code/drasil-example/Drasil/SSP/Body.hs#L201) as well.

**Comment by bmaclach** [comment]
Thanks @Mornix. I don't think I can add `pi_` to the `ChunkDB` because `pi_` is of type `QuantityDict`, which has neither an instance for `Definition` or `ConceptDomain`, both of which are constraints (assuming you meant to add it to the third argument to `cdb`, which is for Concepts.)

**Comment by Mornix** [comment]
I see exactly what you're saying now.

I got the stack trace which shows where the error occurs:
```
ssp: Concept: pi not found in ConceptMap
CallStack (from HasCallStack):
  error, called at ./Language/Drasil/ChunkDB.hs:177:24 in drasil-lang-0.1.54-EH4bOMWUAX3GsbAiVf7pB9:Language.Drasil.ChunkDB
CallStack (from -prof):
  Language.Drasil.ChunkDB.uMapLookup.getFM (Language/Drasil/ChunkDB.hs:177:9-78)
  Language.Drasil.ChunkDB.uMapLookup (Language/Drasil/ChunkDB.hs:(176,1)-(177,78))
  Language.Drasil.ChunkDB.defLookup (Language/Drasil/ChunkDB.hs:189:1-45)
  Language.Drasil.ChunkDB.GetChunk.concpt'.resolve (Language/Drasil/ChunkDB/GetChunk.hs:28:9-47)
  Language.Drasil.ChunkDB.GetChunk.concpt' (Language/Drasil/ChunkDB/GetChunk.hs:(27,1)-(28,47))
  Language.Drasil.ChunkDB.GetChunk.combine' (Language/Drasil/ChunkDB/GetChunk.hs:31:1-53)
  Language.Drasil.ChunkDB.GetChunk.ccss (Language/Drasil/ChunkDB/GetChunk.hs:34:1-74)
  Drasil.SSP.Body.symbTT (Drasil/SSP/Body.hs:214:1-62)
  Drasil.SSP.Body.CAF:symbTT (Drasil/SSP/Body.hs:214:1-6)
```

It looks like `combine'` expects an associated `Concept` when parsing `Expr`s (i.e. a `DefinedQuantityDict` which uses the UID of the underlying `ConceptChunk` as the UID for the entire dqd.) 

A quick grep of data shows that the only `QuantityDict` is `pi_`, "everything else" is a `DefinedQuantityDict`. 

`pi_` being a qd as opposed to a dqd seems like it may be an oversight. 

/cc @JacquesCarette

**Comment by JacquesCarette** [comment]
Yes, `pi_` used to be a much worse hack, and now it is merely a hack. It should be properly defined (as a `DefinedQuantityDict` at minimum, likely more thoroughly than that), then this won't occur anymore.

`ConceptDomain` should probably be 'Math' (or 'Mathematics' or whatever is currently defined). Pick you favourite definition!

**Comment by bmaclach** [comment]
This issue is fixed with the merge of PR #1102. Closing.
